### PR TITLE
Improve event addition API

### PR DESCRIPTION
**Pull request by moosichu**
This PR makes the following changes:
- New Unit Tests
- Hype Date Checking
- Information logging

### CODE REVIEW

File: Sources/HaCWebsiteLib/Controllers/EventApiController.swift [hunk 1]
```diff
@@ -3,26 +3,39 @@ import KituraNet
...
 import DotEnv
+import LoggerAPI
 
 struct EventApiController {
   static var handler: RouterHandler = { request, response, next in
     response.headers["Content-Type"] = "text/plain; charset=utf-8"
     guard let parsedBody = request.body else {
       next()
       response.statusCode = HTTPStatusCode.internalServerError
+      // TODO: Log info about why?
```
Comment (moosichu) [inline Sources/HaCWebsiteLib/Controllers/EventApiController.swift]: The plan is to complete these TODOs before merging, so please make sure that they are all removed (or converted to issues) before any approval!

**Comment by jaredkhan** [comment]
Is there a reason this is all going in one PR? Seems like there's a few independent bits going on here!

**Comment by moosichu** [comment]
Because I was going to implement feature A, but feature B would make A easier. But I couldn't commit feature B as I had already started A. etc. It was just how these ended up developing.

Eg. I added logging as I was debugging stuff. Similarly the environment variable stuff was added because it was needed for the changes being made.

File: Sources/HaCWebsiteLib/Controllers/EventApiController.swift [hunk 2]
```diff
@@ -3,26 +3,39 @@ import KituraNet
...
+      Log.info("Unable to parse the body of the request")
       try response.send("Sorry - we weren't able to parse the body of the request\n").end()
       return
     }
     if case .json(let json) = parsedBody {
       do {
         try saveEvent(json: json)
+        // TODO: Log info about which fields were missing?
```
Comment (moosichu) [inline Sources/HaCWebsiteLib/Controllers/EventApiController.swift]: Reminder: Just delete this TODO

File: Sources/HaCWebsiteLib/Controllers/EventApiController.swift [hunk 3]
```diff
@@ -34,27 +47,29 @@ struct EventApiController {
...
+    // Make sure the event itself falls within the hype period
+    // If it doesn't, we throw an invalidHypePeriod exception
+    if !((hypeStartDate.compare(startDate) == .orderedAscending) &&
+       (startDate.compare(hypeEndDate)   == .orderedAscending) &&
+       (hypeStartDate.compare(endDate)   == .orderedAscending) &&
+       (endDate.compare(hypeEndDate)     == .orderedAscending)) {
+         throw EventParsingError.invalidHypePeriod
+    }
+
+    let location = getOptionalLocation(json: json)
```
Comment (moosichu) [inline Sources/HaCWebsiteLib/Controllers/EventApiController.swift]: Sure this is an optional parameter and should be shuffled down?

File: Sources/HaCWebsiteLib/Controllers/EventApiController.swift [hunk 4]
```diff
@@ -34,27 +43,29 @@ struct EventApiController {
...
     let tags = getOptionalTags(json: json),
     let markdownDescription = json["markdownDescription"] as? String else {
       throw EventParsingError.missingParameters
     }
-    
-    let location = getOptionalLocation(json: json) 
+
+    // Make sure the event itself falls within the hype period
+    // If it doesn't, we throw an invalidHypePeriod exception
+    if !((hypeStartDate.compare(startDate) == .orderedAscending) &&
```
Comment (jaredkhan) [inline Sources/HaCWebsiteLib/Controllers/EventApiController.swift]: Why not use `<`?

File: Sources/HaCWebsiteLib/Middleware/CredentialsServer.swift [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import Credentials
 import CredentialsHTTP
 import DotEnv
+import LoggerAPI
```
Comment (jaredkhan) [inline Sources/HaCWebsiteLib/Middleware/CredentialsServer.swift]: Why are we importing this and then not using it?

**Comment by jaredkhan** [comment]
Have taken a first pass at this. Looks pretty good. A few style comments above

**Comment by jaredkhan** [comment]
@moosichu Can we close this soon please

**Comment by moosichu** [comment]
@jaredkhan review again please.

**Pull request merged**